src/geometry/ArborX_HyperBox.hpp

```diff
@@ -0,0 +1,118 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_HYPERBOX_HPP
+#define ARBORX_HYPERBOX_HPP
+
+#if (__cplusplus >= 201703L) && (KOKKOS_VERSION >= 30699)
```

Comment:
I thought about replacing `__cpluplus` check with `KOKKOS_CXX_STANDARD`, but not sure that's necessary.